src/main/kotlin/org/rust/lang/core/psi/RustMod.kt

```diff
@@ -24,22 +25,19 @@ interface RustMod : RustNamedElement, RustItemsOwner {
 
     val isTopLevelInFile: Boolean
 
-    val canonicalNameInFile: RustQualifiedName?
-
     companion object {
         val MOD_RS = "mod.rs"
     }
 }
 
-/**
- * Returns a string representing a path to this module from the crate root, like `foo::bar::baz`.
- * There is no leading `::`. Returns empty string for the crate root.
- *
- * This name is not guaranteed to be unique: modules from different crates
- * can have the same path within the crate.
- */
-val RustMod.canonicalCratePath: String? get() {
-    if (isCrateRoot) return ""
-    val parentPath = `super`?.canonicalCratePath ?: return null
-    return if (parentPath == "") modName else "$parentPath::$modName"
-}
+val RustMod.canonicalCratePath: RustQualifiedPath?
```

Comment:
```Kotlin
fun quux(mod: RustModItemElement) {
    fun foo(mod: RustMod)         = mod.canonicalCratePath
    fun bar(mod: RustItemElement) = mod.canonicalCratePath

    foo(mod) // This will call a function from `RustMod.kt`
    bar(mod) // This will call a function from `RustItemElement.kt`
}
```